documentation/modules/exploit/windows/http/manageengine_adaudit_plus_authenticated_rce.md

```diff
@@ -0,0 +1,152 @@
+
+This module first runs a few checks to test the provided credentials, retrieve the configured domain(s), and
+obtain the build number of ManageEngine. If the credentials are valid and the target is vulnerable, the module
+creates an alert profile that will be triggered for any failed login attempt to the configured domain.
+
+For versions prior to build 7004, the payload is directly inserted in the custom alert script component of the alert profile.
+
+For builds 7004 and 7005, the module leverages an arbitrary file write vulnerability (CVE-2021-42847) to create a Powershell script
+in the `alert_scripts` directory that contains the payload. Note that this directory will be located under the
+ADAudit Plus installation directory. The name of this script is then provided as the value for the
+custom alert script component of the alert profile. For these builds, Meterpreter payloads such as
+`cmd/windows/powershell/meterpreter/reverse_tcp` do not seem to work and only the `cmd/windows/powershell_reverse_tcp`
+payload has been tested successfully.
+
+This module will automatically delete the created alert profile before completing. This happens even if no shell was obtained.
+
+It should be noted that during a single run, the module will typically authenticate to the target several times.
+This is because ADAudit Plus is very strict about cookies. After a user performs a successful authentication request,
+the server sends a cookie that can be used to visit the dashboard. However, in order to interact with most of the API 
+endpoints, the user must then perform a request to `api/json/configuredDomainsList`. Only then does the server return a
+cookie that can be used to interact with other endpoints. If the above requests are not performed in this exact order,
+or additional requests are performed before the final cookie is obtained, the entire authentication chain needs to be repeated.
+
+This module requires valid credentials for an account with the privileges to create alert scripts.
+It has been successfully tested against ManageEngine ADAudit Plus builds
+[7003](https://archives2.manageengine.com/active-directory-audit/7003/ManageEngine_ADAudit_Plus_x64.exe) and
+[7005](https://archives2.manageengine.com/active-directory-audit/7005/ManageEngine_ADAudit_Plus_x64.exe) running on Windows Server 2012 R2.
+
+Successful exploitation will result in RCE as the user running ManageEngine ADAudit Plus, which will typically be the local administrator.
+
+## Installation Information
+Vulnerable versions of ADAudit Plus are available [here](https://archives2.manageengine.com/active-directory-audit/).
+Versions 7005 and prior are vulnerable by default, so no special configuration is required after installing the application.
+
+After running the installer, you can launch ADAudit Plus by opening a command prompt with administrator privileges
+and then running: `<install_dir>\bin\run.bat`.
```

Comment:
Should mention here that one should not accept the default option to `Start ADAudit Plus Server` after installing. This will not work.

Reply:
Should add more details about how old versions may not allow you to enter into Evaluation mode cause apparently they think that the Evaluation Edition has expired 🙄

Reply:
Weirdly 6077 didn't seem to ask for a license file when I started it via `run.bat`, so perhaps this might not always be the case?

Reply:
ACTUAL STEPS
1. `cd "C:\Program Files\ManageEngine\ADAudit Plus\bin"`
2. `run.bat`
3. Log in to the admin portal. 
4. Default Domain Controllers Policy not configured for daforest.com <- Click on the Configure link on this.
5. Go to notifications, "Product Not Installed As Service" and click "Install Now"
6. Open Group Policy Management on the domain controller and go to Forest->Domains->Select your domain->Default Domain Policy and right click on it then click Edit.
7. Computer Configuration->Policies->Windows Settings->Security Settings->Advanced Audit Policy Configuration->Audit Policies->Logon/Logoff and set `Audit Logoff`, `Audit Logon` and `Audit Special Logon` and `Audit Other Logon/Logoff Events` and set them to `Configure the following audit events` and check that box and `Success` and `Failure`.
8. Log off
9. Should be able to run the module now.

Reply:
Phew...okay that works. Lets update the documentation with this and then I think we should be good.